guide/samples/tests/driver.rs

```diff
@@ -91,26 +87,49 @@ mod driver {
         // The sample will delete this secret. If that fails, the cleanup step
         // for the integration tests will garbage collect it in a couple of
         // days.
-        let _ = client
-            .create_secret(format!("projects/{project_id}"))
+        let _ = user_guide_samples::error_handling::create_secret(&client, &project_id, &secret_id)
+            .await?;
+        user_guide_samples::retry_policies::request_retry(&client, &project_id, &secret_id).await
+    }
+
+    #[tokio::test]
+    async fn error_handling() -> user_guide_samples::Result<()> {
+        use google_cloud_gax::retry_policy::AlwaysRetry;
+        use google_cloud_gax::retry_policy::RetryPolicyExt;
+        use google_cloud_secretmanager_v1 as sm;
+        use std::time::Duration;
+
+        let project_id = std::env::var("GOOGLE_CLOUD_PROJECT").unwrap();
+        let secret_id: String = rand::rng()
+            .sample_iter(&Alphanumeric)
+            .take(SECRET_ID_LENGTH)
+            .map(char::from)
+            .collect();
+
+        let client = sm::client::SecretManagerService::builder()
             .with_retry_policy(
                 AlwaysRetry
                     .with_attempt_limit(5)
                     .with_time_limit(Duration::from_secs(15)),
             )
-            .set_secret_id(&secret_id)
-            .set_secret(
-                sm::model::Secret::new()
-                    .set_replication(sm::model::Replication::new().set_replication(
-                        sm::model::replication::Replication::Automatic(
-                            sm::model::replication::Automatic::new().into(),
-                        ),
-                    ))
-                    .set_labels([("integration-test", "true")]),
-            )
+            .build()
+            .await?;
+        // The secret is immediately deleted. If that fails, the cleanup step
+        // for the integration tests will garbage collect it in a couple of
+        // days.
+        let _ = user_guide_samples::error_handling::create_secret(&client, &project_id, &secret_id)
+            .await?;
```

Comment:
I would prefer to remove this and test the not found branch

Reply:
Why not both?